### PR TITLE
Update link as current Devdoc Magento version

### PR DESCRIPTION
**Pull request by magesanjay**
## This PR is a:

- Update link as per current magento version of devdoc.

## Summary

When this pull request is merged, it will Update link as per current magento version of devdoc.

Below are the lines which have updated links on text "JavaScript component" and "accordion".
- Insert a `JavaScript component` in .phtml page templates.
- To call a widget in JS code, use a notation similar to the following (`accordion` widget is intiialized on the [data-role=example] element as illustration)

## Additional information

List all affected URLs 

1. https://devdocs.magento.com/guides/v2.1/javascript-dev-guide/javascript/js_init.html
2. https://devdocs.magento.com/guides/v2.2/javascript-dev-guide/javascript/js_init.html
3. https://devdocs.magento.com/guides/v2.3/javascript-dev-guide/javascript/js_init.html

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

**Comment by jeff-matthews** [comment]
@magesanjay, are there any differences between each version of the `javascript-dev-guide/javascript/js_init.md` file? I haven't done a diff to check, but I will to make sure.

If there is no difference between version 2.1, 2.2, or 2.3, then we should leave the symlinks and replace the `{{ site.gdeurl}}` variable with the `{{ page.baseurl }}` variable.

For example:

```Insert a [JavaScript component]({{ page.baseurl }}/javascript-dev-guide/bk-javascript-dev-guide.html#js_terms) in `.phtml` page templates.```

AND

```To call a widget in JS code, use a notation similar to the following ([accordion]({{ page.baseurl }}frontend-dev-guide/javascript/widget_accordion.html) widget is intiialized on the `[data-role=example]` element as illustration):```

**Comment by magesanjay** [comment]
@jeff-matthews, No there is not any other changes between each version.
I have replaced `{{ site.gdeurl }}` variable with the `{{ page.baseurl }}` variable.
Can you please check and confirm it.

**Comment by jeff-matthews** [comment]
Perfect! Thanks @magesanjay.

**Comment by jeff-matthews** [comment]
running tests

**Comment by jeff-matthews** [comment]
Link checking failed:

```
+ rake test:cicd
[35mChecking links with htmlproofer...[0m
Running ["LinkChecker::DoubleSlashCheck", "ImageCheck", "LinkCheck", "ScriptCheck"] on ["_site"] on *.html... 


Ran on 3824 files!


- internally linking to /guides/v2.1frontend-dev-guide/javascript/widget_accordion.html, which does not exist
  *  _site/guides/v2.1/javascript-dev-guide/javascript/js_init.html (line 2259)
     <a href="/guides/v2.1frontend-dev-guide/javascript/widget_accordion.html">accordion</a>
- internally linking to /guides/v2.1javascript-dev-guide/bk-javascript-dev-guide.html#js_terms, which does not exist
  *  _site/guides/v2.1/javascript-dev-guide/javascript/js_init.html (line 2139)
     <a href="/guides/v2.1javascript-dev-guide/bk-javascript-dev-guide.html#js_terms">JavaScript component</a>
- internally linking to /guides/v2.2frontend-dev-guide/javascript/widget_accordion.html, which does not exist
  *  _site/guides/v2.2/javascript-dev-guide/javascript/js_init.html (line 2321)
     <a href="/guides/v2.2frontend-dev-guide/javascript/widget_accordion.html">accordion</a>
- internally linking to /guides/v2.2javascript-dev-guide/bk-javascript-dev-guide.html#js_terms, which does not exist
  *  _site/guides/v2.2/javascript-dev-guide/javascript/js_init.html (line 2201)
     <a href="/guides/v2.2javascript-dev-guide/bk-javascript-dev-guide.html#js_terms">JavaScript component</a>
- internally linking to /guides/v2.3frontend-dev-guide/javascript/widget_accordion.html, which does not exist
  *  _site/guides/v2.3/javascript-dev-guide/javascript/js_init.html (line 2377)
     <a href="/guides/v2.3frontend-dev-guide/javascript/widget_accordion.html">accordion</a>
- internally linking to /guides/v2.3javascript-dev-guide/bk-javascript-dev-guide.html#js_terms, which does not exist
  *  _site/guides/v2.3/javascript-dev-guide/javascript/js_init.html (line 2257)
     <a href="/guides/v2.3javascript-dev-guide/bk-javascript-dev-guide.html#js_terms">JavaScript component</a>
- trying to find hash of /guides/v2.1javascript-dev-guide/bk-javascript-dev-guide.html#js_terms, but /apps/jenkins/latest/workspace/devdocs-test/_site/guides/v2.1javascript-dev-guide/bk-javascript-dev-guide.html does not exist
  *  _site/guides/v2.1/javascript-dev-guide/javascript/js_init.html (line 2139)
     <a href="/guides/v2.1javascript-dev-guide/bk-javascript-dev-guide.html#js_terms">JavaScript component</a>
- trying to find hash of /guides/v2.2javascript-dev-guide/bk-javascript-dev-guide.html#js_terms, but /apps/jenkins/latest/workspace/devdocs-test/_site/guides/v2.2javascript-dev-guide/bk-javascript-dev-guide.html does not exist
  *  _site/guides/v2.2/javascript-dev-guide/javascript/js_init.html (line 2201)
     <a href="/guides/v2.2javascript-dev-guide/bk-javascript-dev-guide.html#js_terms">JavaScript component</a>
- trying to find hash of /guides/v2.3javascript-dev-guide/bk-javascript-dev-guide.html#js_terms, but /apps/jenkins/latest/workspace/devdocs-test/_site/guides/v2.3javascript-dev-guide/bk-javascript-dev-guide.html does not exist
  *  _site/guides/v2.3/javascript-dev-guide/javascript/js_init.html (line 2257)
     <a href="/guides/v2.3javascript-dev-guide/bk-javascript-dev-guide.html#js_terms">JavaScript component</a>
rake aborted!
HTML-Proofer found 9 failures!```

File: guides/v2.1/javascript-dev-guide/javascript/js_init.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ redirect_from:
 
 This topic describes different ways to call and initialize JavaScript in Magento 2:
 
-- Insert a [JavaScript component]({{ site.gdeurl }}javascript-dev-guide/bk-javascript-dev-guide.html#js_terms) in `.phtml` page templates.
+- Insert a [JavaScript component]({{ page.baseurl }}javascript-dev-guide/bk-javascript-dev-guide.html#js_terms) in `.phtml` page templates.
```
Comment (jeff-matthews) [inline guides/v2.1/javascript-dev-guide/javascript/js_init.md]:
```suggestion
- Insert a [JavaScript component]({{ page.baseurl }}/javascript-dev-guide/bk-javascript-dev-guide.html#js_terms) in `.phtml` page templates.
```

File: guides/v2.1/javascript-dev-guide/javascript/js_init.md [hunk 2]
```diff
@@ -128,7 +128,7 @@ require([
 
 ## Calling JS components requiring initialization in JS files {#js_widget_init}
 
-To call a widget in JS code, use a notation similar to the following ([accordion]({{ site.gdeurl }}frontend-dev-guide/javascript/widget_accordion.html) widget is intiialized on the `[data-role=example]` element as illustration):
+To call a widget in JS code, use a notation similar to the following ([accordion]({{ page.baseurl }}frontend-dev-guide/javascript/widget_accordion.html) widget is intiialized on the `[data-role=example]` element as illustration):
```
Comment (jeff-matthews) [inline guides/v2.1/javascript-dev-guide/javascript/js_init.md]:
```suggestion
To call a widget in JS code, use a notation similar to the following ([accordion]({{ page.baseurl }}/frontend-dev-guide/javascript/widget_accordion.html) widget is intiialized on the `[data-role=example]` element as illustration):
```

**Comment by magesanjay** [comment]
@jeff-matthews I have added suggestion.

**Comment by jeff-matthews** [comment]
running tests

**Comment by jeff-matthews** [comment]
Link checking failed again:

```
+ rake test:cicd
[35mChecking links with htmlproofer...[0m
Running ["LinkChecker::DoubleSlashCheck", "ImageCheck", "ScriptCheck", "LinkCheck"] on ["_site"] on *.html... 


Ran on 3824 files!


- internally linking to /guides/v2.1/frontend-dev-guide/javascript/widget_accordion.html, which does not exist
  *  _site/guides/v2.1/javascript-dev-guide/javascript/js_init.html (line 2259)
     <a href="/guides/v2.1/frontend-dev-guide/javascript/widget_accordion.html">accordion</a>
- internally linking to /guides/v2.2/frontend-dev-guide/javascript/widget_accordion.html, which does not exist
  *  _site/guides/v2.2/javascript-dev-guide/javascript/js_init.html (line 2321)
     <a href="/guides/v2.2/frontend-dev-guide/javascript/widget_accordion.html">accordion</a>
- internally linking to /guides/v2.3/frontend-dev-guide/javascript/widget_accordion.html, which does not exist
  *  _site/guides/v2.3/javascript-dev-guide/javascript/js_init.html (line 2377)
     <a href="/guides/v2.3/frontend-dev-guide/javascript/widget_accordion.html">accordion</a>
rake aborted!
HTML-Proofer found 3 failures!```

File: guides/v2.1/javascript-dev-guide/javascript/js_init.md [hunk 3]
```diff
@@ -128,7 +128,7 @@ require([
 
 ## Calling JS components requiring initialization in JS files {#js_widget_init}
 
-To call a widget in JS code, use a notation similar to the following ([accordion]({{ site.gdeurl }}frontend-dev-guide/javascript/widget_accordion.html) widget is intiialized on the `[data-role=example]` element as illustration):
+To call a widget in JS code, use a notation similar to the following ([accordion]({{ page.baseurl }}/frontend-dev-guide/javascript/widget_accordion.html) widget is intiialized on the `[data-role=example]` element as illustration):
```
Comment (jeff-matthews) [inline guides/v2.1/javascript-dev-guide/javascript/js_init.md]: This file doesn't exist in the `frontend-dev-guide`. The correct path is:

```javascript-dev-guide/widgets/widget_accordion.html```

**Comment by magesanjay** [comment]
@jeff-matthews I have corrected url paths.

**Comment by jeff-matthews** [comment]
running tests

**Comment by jeff-matthews** [comment]
Tests are still failing

```
+ rake test:cicd
[35mChecking links with htmlproofer...[0m
Running ["LinkChecker::DoubleSlashCheck", "ImageCheck", "LinkCheck", "ScriptCheck"] on ["_site"] on *.html... 


Ran on 3824 files!


- internally linking to /guides/v2.1/javascript-dev-guide/javascript/widget_accordion.html, which does not exist
  *  _site/guides/v2.1/javascript-dev-guide/javascript/js_init.html (line 2259)
     <a href="/guides/v2.1/javascript-dev-guide/javascript/widget_accordion.html">accordion</a>
- internally linking to /guides/v2.2/javascript-dev-guide/javascript/widget_accordion.html, which does not exist
  *  _site/guides/v2.2/javascript-dev-guide/javascript/js_init.html (line 2321)
     <a href="/guides/v2.2/javascript-dev-guide/javascript/widget_accordion.html">accordion</a>
- internally linking to /guides/v2.3/javascript-dev-guide/javascript/widget_accordion.html, which does not exist
  *  _site/guides/v2.3/javascript-dev-guide/javascript/js_init.html (line 2377)
     <a href="/guides/v2.3/javascript-dev-guide/javascript/widget_accordion.html">accordion</a>
rake aborted!
HTML-Proofer found 3 failures!```

**Comment by magesanjay** [comment]
Hi @jeff-matthews i have update link now.

**Comment by jeff-matthews** [comment]
running tests

**Pull request merged**